pkgs/development/tools/godot/3/mono/default.nix

```diff
@@ -0,0 +1,69 @@
+{ godot
+, callPackage
+, mkNugetDeps
+, mkNugetSource
+, mono
+, dotnet-sdk
+, writeText
+}:
+
+godot.overrideAttrs (self: base: {
+  pname = "godot-mono";
+
+  buildDescription = "mono build";
+
+  nativeBuildInputs = base.nativeBuildInputs ++ [ mono dotnet-sdk ];
+
+  glue = callPackage ./glue.nix {};
+
+  nugetDeps = mkNugetDeps { name = "deps"; nugetDeps = import ./deps.nix; };
+
+  nugetSource =
+    mkNugetSource {
+      name = "${self.pname}-nuget-source";
+      description = "A Nuget source with dependencies for ${self.pname}";
+      deps = [ self.nugetDeps ];
+    };
+
+  nugetConfig = writeText "NuGet.Config" ''
+    <?xml version="1.0" encoding="utf-8"?>
+    <configuration>
+      <packageSources>
+        <add key="${self.pname}-deps" value="${self.nugetSource}/lib" />
+      </packageSources>
+    </configuration>
+  '';
+
+  sconsFlags = base.sconsFlags ++ [
+    "module_mono_enabled=true"
+    "mono_prefix=${mono}"
+  ];
+
+  shouldConfigureNuget = true;
```

Comment:
Probably out of the loop but why do we need this at all in make-dips? It's the only usage of it being overridden to false that I can see.

Reply:
To "configure Nuget" entails building a local nuget repo in the nix store containing all the nuget packages needed for the mono build of godot, and dropping a config file referencing it into a standard location within a temporary home folder.  To build Godot mono, we always want to do that.

This is built using deps.nix which specifies exactly what packages, their versions, and their hashes.

But whenever we update to a newer version of godot, it may depend on different packages or versions thereof, we need to rebuild deps.nix.  This is done by building godot mono *without* the nuget config in place, causing msbuild to go out to the internet and download nuget packages directly, and then to generate deps.nix based on what it downloads.  This cannot be done by nix build, because msbuild fetching nuget packages can't be done in a sandbox.  Thus, we perform this build within a nix shell.  Except here, we want to generate a new nuget config so we do *not* want to "configure Nuget", hence `shouldConfigureNuget = false;`.

Reply:
Ahhhh, that makes sense, thanks for clearing this up!